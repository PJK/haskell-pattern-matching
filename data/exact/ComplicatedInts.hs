comp :: Int -> Int
comp x
    | x > 5 = 1
    | x < 0 = 1
comp x
    | x < 3 = 1
    | x > 3 = 1

