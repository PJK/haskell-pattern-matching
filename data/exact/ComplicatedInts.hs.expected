{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "comp",
                {
                    "tag": "NonExhaustive",
                    "contents": [
                        [
                            [
                                {
                                    "tag": "VariablePattern",
                                    "contents": "~a"
                                }
                            ],
                            "Constraints:\n  ~l == False\n  ~l == ~a > 3\n  ~i == False\n  ~i == ~a < 3\n  ~f == False\n  ~f == ~a < 0\n  ~c == False\n  ~c == ~a > 5\n\nSatisfiable. Model:\n  ~l = False :: Bool\n  ~a =     3 :: Integer\n  ~i = False :: Bool\n  ~f = False :: Bool\n  ~c = False :: Bool\n"
                        ]
                    ]
                }
            ]
        ],
        [
            [
                "comp",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
